plugins/out_s3/s3.c

```diff
@@ -1540,7 +1547,14 @@ static struct multipart_upload *create_upload(struct flb_s3 *ctx,
     m_upload->tag = tmp_sds;
     m_upload->upload_state = MULTIPART_UPLOAD_STATE_NOT_CREATED;
     m_upload->part_number = 1;
-    m_upload->init_time = time(NULL);
+    if (timep == NULL) {
+        m_upload->init_time = time(NULL);
+        flb_plg_error(ctx->ins, "Failed to get timestamp from the log record "
+                      "used for the S3 key, use the current time instead.");
+    }
+    else {
+        m_upload->init_time = timep;
+    }
```

Comment:
Why do you set init_time to timep which is the first_log_timestamp?
This will cause more timeout related errors see
https://github.com/fluent/fluent-bit/blob/760956f50cdc7eef3047df20d2299202a8c68594/plugins/out_s3/s3.c#L2161-L2165

Reply:
return to origin